### ISSUE TITLE
Weekly MAVLink call and coordination

### ISSUE THREAD

**Issue by LorenzMeier** [mavlink/mavlink#888]
Hi All,

Given the increased interest in standardization and MAVLink picking up more and more "micro-services" like the camera control standard, I believe a regular call would help. This is in particular necessary to coordinate efforts across different open source projects or vendors.

I would also like to facilitate the collaboration between different teams / groups and find common ground focused around technical merit. Calls and in-person meetings are essential to create the cohesion required for this.

I would propose a bi-weekly call, alternating between 09:00 a.m. CET / 15:00h Beijing / 17:00h Sydney time and 18:00 p.m. CET / 9 a.m. PST every 2nd Wednesday of the month.

I would also like to find a time for interested parties to meet at AUVSI Xponential early May in person.

Please leave feedback / suggestions in this thread.

**Comment by ashirsat** [comment]
@LorenzMeier it would also be a good idea to have some sort of minutes or summary for the meeting that can be posted online so that if people miss the meeting they can get the updates

**Comment by hamishwillee** [comment]
@amilcarlucas Would be great to have your involvement.

**Comment by amilcarlucas** [comment]
Yes, I would like to participate

**Comment by WickedShell** [comment]
I should be able to make either of the call times, and I will be at AUVSI as well.

**Comment by keenanjohnson** [comment]
I would also be very glad to join the conversation and I'll also be around during AUVSI.

**Comment by auturgy** [comment]
I won't be at AUVSI but am glad to see this effort.

**Comment by LorenzMeier** [comment]
Thanks all for the feedback! I think I left enough time to comment for anybody interested initially - any comments on the times I proposed? Are they sensible or do we need something like a Doodle poll?

**Comment by TSC21** [comment]
I think a Doodle would be a good option yes.

**Comment by mrpollo** [comment]
Please join us for our first call on Wednesday, April 25th 2018, more info here https://mavlink.io/en/contributing/contributing.html#support

**Everyone is welcome!**

**Comment by auturgy** [comment]
I think the first meeting needs an agreed agenda prior to scheduling: developing the framework and ground rules seems to me to be critical to success, and needs to be done early. Arbitrarily determining a time without determining where the key players are is not a good start.

**Comment by mrpollo** [comment]
Thanks for the feedback @auturgy, we are trying to find the "key players," if this time doesn't work for you or others, please tell us what does work, and we will work it out as a group. I agree on the agenda; I'll post back with a link to contribute.

Everyone, this meeting should be seen as a "first step," we aren't claiming this first meeting is going to solve every problem for everyone, it will take **time, effort and coordination** to get this project where we all want it to be, let us get there together.

**Comment by WickedShell** [comment]
I'm confused by what happened to the scheduling of this call? It was originally proposed as a biweekly call, (or every second Wednesday which would make it a monthly call) with alternating times to help other timezones. A weekly call seems like to be more frequent then is needed, but I'm mostly looking to clarify where the timing changed/target frequency.

**Comment by keenanjohnson** [comment]
Is there going to be a meeting at AUVSI?

**Comment by auturgy** [comment]
So it seems this meeting occurred without an agenda being sent out, or any real attempt to put this in the view of the largest mavlink users.
Can I ask that when this is rescheduled, the agenda be distributed prior, and notifications be sent not only to Slack, but gitter and discuss.ardupilot.org

**Comment by LorenzMeier** [comment]
There was some confusion on the call - I specifically wanted to set up a Doodle so that we can find times that are non-discriminative. Consequently the call Ramon proposed was not an official MAVLink dev call and specs / direction were not discussed.

Here is the proper poll for the first official call:
https://doodle.com/poll/qmmprzpnr6z2hib4

Goal is to start in two weeks and then keep a weekly schedule on two time slots.

**Comment by LorenzMeier** [comment]
In terms of AUVSI XPONENTIAL Meetup: Here is another Doodle Poll for this:

https://doodle.com/poll/2gmtxcipv2ghhkup

**Comment by jacobsenmd** [comment]
Any chance we can get this nailed down? Thanks!

**Comment by magicrub** [comment]
Time is Tuesday, May 1st from 2:00pm to 3:30pm.

**Comment by LorenzMeier** [comment]
Just to make sure there is no confusion on location: Lobby of the Hyatt near the conference center (straight ahead of the main entrance)

**Comment by mrpollo** [comment]
To everyone who attended, can we please get a summary of what was discussed and the outcome? Would love to hear how the first meeting went.

**Comment by amilcarlucas** [comment]
When will the the meeting take place ??

**Comment by LorenzMeier** [comment]
I just returned from travel, sorry about the delay on commenting here. I will provide feedback on the in-person meeting and coordinate the Doodle results with you.

**Comment by magicrub** [comment]
Thanks @LorenzMeier

**Comment by LorenzMeier** [comment]
Wednesdays 6-7 p.m Zurich / 9-10 a.m. PST seems to fit for most participants, so I would suggest starting with that on June 6, 2018.

Link to poll again: https://doodle.com/poll/qmmprzpnr6z2hib4

Meeting notes:

We had five attendees total from the APM and PX4 dev teams and discussed:

 - Coordinating some low-hanging fruits:
  - Well-defined behaviour of home position
  - Well-defined behaviour of MAVLink stream rates
 - Syncing the required fixes upstream and then getting rid of the pymavlink fork: https://github.com/mavlink/pymavlink
 - Setting up a weekly call to coordinate - the conclusion was that all interested participants are either based in Europe or the US
 - Working towards setting up a test suite to make it easier for implementers to comply with the spec
 - Establishing minimal base profiles that clearly define which messages and which micro-services have to be implemented
 - Auditing the language support such that we have clearly supported language bindings and "non-supported" ones that might not be as stable.

**Comment by WickedShell** [comment]
I will not be able to make meetings at that time, but am interested in the outcome and look forward to the result.

**Comment by auturgy** [comment]
Likewise (2am for me).

Regards,

James

**Comment by hamishwillee** [comment]
Would it be beneficial/could the meeting be held at another time on a regular but reduced rate to pick up some of the other interested parties? E.g. one week in 4.  Just an idea.

**Comment by OXINARF** [comment]
From the beginning it has always been the idea to have two alternating times because of the worldwide reach of MAVLink; I'm not sure how, for two times, we've ended up with a single meeting time.

**Comment by auturgy** [comment]
The first poll was pretty clear about selecting two time slots. I’m not sure how that got confused.

Regards,

James

**Comment by mrpollo** [comment]
There's not a clear winner that I can see from the remainder of timeslots for the first poll, the only one that caught my attention was the Tuesday @ 7:00 am UTC with 4 participants, I'm ok with scheduling whatever the community decides, I just don't want to guess.

**Comment by hamishwillee** [comment]
@mrpollo Looking at the original poll, @auturgy , Michael and Thomas prefer 5-6pm Australia time. The day follows pattern Mon/Tues/Weds in subsequent weeks - I expect that this was somewhat random and they could probably all do any of those days - correct?

So perhaps pick in that timeslot and find a day that someone from the first group is also able to attend (to aid co-ordination)?

**Comment by auturgy** [comment]
Thanks.
Can we kill the second poll? The way it is, likely to create more noise than signal.
And don’t forget to pick a time yourself Hamish!

**Comment by hamishwillee** [comment]
I can do with 5-6pm on any day except (currently) Friday, and I'll fit in with what helps the team. 

I don't want to hijack the discussion until we've resolved a date. But once that is done I'm keen to better understand how "we keep vigilant both groups coordinate" (thanks Ramon). Specifically the mechanisms for coordinating issues that need more than just one group to agree.

**Comment by LorenzMeier** [comment]
How about 5-6 p.m. Wednesdays, initially weekly, potentially bi-weekly? I've blocked that for now in my calendar so if its ok I could start pretty much any time.

**Comment by hamishwillee** [comment]
> How about 5-6 p.m. Wednesdays, initially weekly, potentially bi-weekly? I've blocked that for now in my calendar so if its ok I could start pretty much any time.

Works for me! Even better, it means that key members of the community will be present in both meetings.  Ramon indicated he would try to attend in this timezone too.

**Comment by auturgy** [comment]
Works for me too

Regards,

James

**Comment by WickedShell** [comment]
What timezone is that 5-6PM in?

BTW the post here: https://github.com/mavlink/mavlink/issues/888#issuecomment-392868893 states 4PM UTC at the moment, the calendar link states 3PM UTC.

**Comment by hamishwillee** [comment]
> What timezone is that 5-6PM in?

AEST - Melbourne/Sydney time

**Comment by dagar** [comment]
Call in progress - https://zoom.us/meeting/579545140/ics?icsToken=fb96681cf4458be9f1f85694037a3cc7f358d5a9479ec0a06dedc267e49649ac

**Comment by magicrub** [comment]
Aww crap. I missed the call and can't do the next time zone one. Next week is better. I talked to Michael dB and he says next week is better too. Sorry!

**Comment by auturgy** [comment]
Thanks, appreciated.

Regards,

James

**Comment by hamishwillee** [comment]
Hi Everyone, just attended the MAVLink meeting 5pm AEST - no one but me and Lorenz there for most of it. We took the opportunity to discuss documentation: https://px4.slack.com/archives/C52DXKYLV/p1528875107000403

WOuld have been good if we could close out some open issues on this system. Was there something wrong with the notification?

**Comment by auturgy** [comment]
Nothing wrong with the notification on your end afaict.
Seems I had a calendar sync issue at mine though.
Can help with the docs if you like.

Regards,

James

**Comment by WickedShell** [comment]
I suffered a badly delayed flight and was driving home (without reliable phone service) through the call. Should not be a future issue/ongoing issue.

**Comment by magicrub** [comment]
uhm, I added the two meetings from @mrpollo and none just occured.

This is getting ridiculous

**Comment by magicrub** [comment]
Topics from the latest meeting:
- pymavlink fork needs to be solved, deferred to next meeting
- using common.xml as without a dialect. (opposite model of ArduPilot)
- Start thinking about MAVLink v3 to become the protocol to rule them all (long term goal)
- create a CI test that all autopilots and GCSs must test against to be compliant and to ensure they stay that way in the future.
- MAVLink project, independent of any autopilot, must dictate stricter usage rules (example: describe how to set home or mission sync).
- how to solve the old-msg-trash. Version. Deprecation policy.
- Extensions are broken because it allows adding extra ones and you don't know if they're populated or not (always zero). We need a better trash collection scheme.
- WIPs should not be allowed: Proposed solution is to move any new ones to special file that is not normally included such as WIP/Experimental/RFC.xml file
- get stakeholders to attend.

**Comment by amilcarlucas** [comment]
Can someone please post the meeting times in UTC?
There is one in the evening and one in the morning right?

**Comment by mrpollo** [comment]
Call times in UTC

Call 1: 7:00am UTC
Call 2: 4:00pm UTC

You can see the calendar and subscribe to any meetings here http://dev.px4.io/en/contribute/#calendar

**Comment by mrpollo** [comment]
Just a heads up, the mavlink call is bi-weekly, we don't have a call scheduled this week (W25).

**Comment by hamishwillee** [comment]
MAVLink meetings are scheduled for today. I plan on attending the one at 5pm AEST for first half hour (got another meeting at 5.30). 
- Who else is going?
- Does anyone have any specific issues for the agenda? What I would like to discuss is:
  - We had a few new issues/suggestions that did not get responses this week. Would be good to get some eyes on them - e.g.  https://github.com/mavlink/mavlink/issues/934
  - More discussion on things like versioning for which we had a flurry of activity but have not progressed.

**Comment by auturgy** [comment]
I’ll be on the 5pm AEST call 

Regards,

James

**Comment by hamishwillee** [comment]
MAVLink meeting Australia. I had to duck out early, but we covered:
1. [RFC9](https://github.com/mavlink/rfcs/pull/9)
   - not sure why system time can't be used. That may not always be correct, but will always be consistent for the vehicle in terms of linear time, across reboots.
   - need to understand whether this can be added as MAVLink2 (prefer not to add in MAVLink1)
1. Java Parser
   - Current implementation requires changes that will mandate breaking changes. 
   - Suggestions of changes have been met with silence. Impossible to progress without community interaction.
    - not clear if right people are seeing the discussions
    - stakeholders not clear - who is using parser, who can approve changes, who can contribute to parser
    - General feeling seemed to be that making breaking changes needed, so we need to identify stakeholders who can contribute/agree, then work on timing, notifications etc.
   - ACTION: @thomasgubler To take to next meeting to see if we have any interested parties.
1. We need process for sharing discussions across the two MAVLink meetings and between weeks.
    - General support for having a clear agenda and a scratchpad.
    - **Action** @mrpollo to set up google doc(s) for these in the Dronecode google drive. Share edit link with team for the scratchpad. [Don't you love getting agenda items from meetings you don't attend]

**Comment by amilcarlucas** [comment]
Sorry could not attend. Will have to wait two more weeks :(

**Comment by mrpollo** [comment]
@hamishwillee I think google docs is too restrictive for sharing minutes, my proposal is to use the [discourse forum](http://discuss.px4.io/c/mavlink)

2nd MAVLink call had some problems, Slack and Zoom where down for a while this morning, Liam, Thomas and myself where on the call.

**Comment by auturgy** [comment]
I think google doc is better, and more bipartisan.
People shouldn’t need a login to your user forum to participate.

Regards,

James

**Comment by mrpollo** [comment]
I have no problem with that @auturgy, but whoever is taking minutes on each call should create the doc and share access to the group, we still need to post those minutes somewhere for distribution and archiving.

**Comment by dagar** [comment]
I don't feel that strongly about meeting minutes, but generally google docs isn't great for visibility. Over time the links end up lost or hard to find. If you post the content somewhere like discuss it ends up indexed by google.

I don't really see the problem with Dronecode discuss (you can login via github), but if it's not acceptable then I wouldn't be opposed to just dumping the notes straight to a github issue.

**Comment by thomasgubler** [comment]
> I don't really see the problem with Dronecode discuss (you can login via github), but if it's not acceptable then I wouldn't be opposed to just dumping the notes straight to a github issue.

https://github.com/mavlink/mavlink/wiki  ?

**Comment by mrpollo** [comment]
Whatever the maintainer team feels most comfortable with is fine with me, I don't have any preference, GitHub wiki works. To summarize the proposal is to use google docs and host the links on mavlink/mavlink wiki. Is anyone else in favor of this?  (or against).

**Comment by hamishwillee** [comment]
Sounds good to me. 
- mavlink wiki for the agenda, linked to google docs for sharing discussion between the two meetings
- Might be nice to have the minutes summarised after the agenda, but that would depend on someone being happy to do that work.

**Comment by mrpollo** [comment]
Proposal for discussion topics on next week calls:

- Maintainers, responsibilities and overall project governance.
- Meeting health, meeting minutes,  attendance, agendas, and where to publish
- Cross-call synergy.

I would love to hear more proposals from the interested parties, we have 1 week to come up with a good agenda, please don't wait until the very last minute.

**Comment by hamishwillee** [comment]
Sounds like a good set. I've created that first Agenda post [here](https://github.com/mavlink/mavlink/wiki/20180711-Weekly-Meeting), seeded with above ideas.  People can add their ideas there if they prefer.

**Comment by hamishwillee** [comment]
Thanks for meeting yesterday. I've added agenda for next week in wiki: https://github.com/mavlink/mavlink/wiki/20180725-Weekly-Meeting

@thomasgubler in the Meeting 2 the notes were

- main issue: need maintainers with clear responsibilities
- idea: committee 7 people, self nominated

Did you guys take a shot at guessing what you think those responsibilities of the above might be? (maintainers/committee members)?

- what is mavlink: is it language definitions or also a reference implementation

Did you come to conclusions on this?
IMO it is: 
- over the wire package format, 
- sub-protocol definitions
- minimum set of messages to support sub protocol definitions
- And this is what makes "common.xml"
- documentation
- test suite that validates implementations

I'd argue that while the generators/reference implementation in theory need not strictly speaking be part of the protocol, in practice 
- if you don't have a reference it is very hard to validate that the assumptions behind the protocol design work.
- The reference implementation currently is the documentation of some aspects of the design
- It is needed to improve adoption. I guarantee that some people wouldn't use MAVLink if they had to write their own generator etc.

> What about pymavlink fork

Also, was the state of the pymavlink fork discussed?

**Comment by hamishwillee** [comment]
MAVLink Meeting today! https://github.com/mavlink/mavlink/wiki/20180725-Weekly-Meeting

I hope all with actions are ready to discuss them.

**Comment by amilcarlucas** [comment]
Am I correct to assume that the meeting will take place in 7 hours?

**Comment by hamishwillee** [comment]
@amilcarlucas The dronecode calendar is linked near the top of this [doc](https://github.com/mavlink/mavlink/wiki ), and it says about 7 hours from right now. YOu should check that again!

Please update the [Agenda here](https://github.com/mavlink/mavlink/wiki/20180725-Weekly-Meeting) with any issues you particularly want unblocked.

**Comment by hamishwillee** [comment]
@amilcarlucas 

You are a legend.

I just noticed that mavlink/mavlink has now switched to using the ArduPilot/pymavlink fork. This should make our lives a lot easier - and would not have been possible if you had not invested so much of your time in getting the two pymavlink forks in sync.

Thank you!
